### ISSUE TITLE
Uses of the word `export` get removed from code blocks

### ISSUE THREAD

**Issue by joefiorini** [ChristopherBiscardi/gatsby-mdx#300]
**Describe the bug**
I have a code block where I am showing an example that uses named & default exports. The word `export` is getting removed (most likely due to this code: https://github.com/ChristopherBiscardi/gatsby-mdx/blob/master/packages/gatsby-mdx/utils/gen-mdx.js#L129-L130).

**To Reproduce**
This is very easy to reproduce by creating a code block, adding some code and having an export that is not on the first line.

E.g. From my blog:
```markdown
```typescript
// Alert.tsx

export type AlertStatus = 'not_shown' | 'success' | 'error' | 'dismissed'

function Alert({ status, message }: { status: AlertStatus; message: string }) {
  return <div className={`alert alert--${status}`}>{message}</div>
}

export default Alert
```

(Could not figure out how to show a Markdown code block inside a Markdown code block)

This gets converted to:

```
type AlertStatus = 'not_shown' | 'success' | 'error' | 'dismissed'

function Alert({ status, message }: { status: AlertStatus; message: string }) {
  return <div className={`alert alert--${status}`}>{message}</div>
}

default Alert
```

**Expected behavior**
I expect MDX to leave code inside code blocks alone (or have a documented way to escape keywords) so that `export` statements in examples don't get replaced.

**Additional context**
At a glance this is similar to #184 & #212, although brought up in comments on #184 the core issue here is quite different than the one in those.

**Comment by ChristopherBiscardi** [comment]
I'd accept a PR to fix this, for sure.

> (Could not figure out how to show a Markdown code block inside a Markdown code block)

Use 4 spaces before each line on the code block, like:

    ```md
    # some Markdown
    ```

**Comment by johno** [comment]
This should be fixed in gatsby-mdx@^0.6 and @mdx-js/mdx@1
